devito/finite_differences/tools.py

```diff
@@ -261,3 +261,22 @@ def generate_indices_staggered(func, dim, order, side=None, x0=None):
             ind = [start, start - diff]
 
     return start, tuple(ind)
+
+
+def make_get_shifted_oringin(shift, shape):
+    """
+    Returns a callable that calculates a shifted origin for each derivative
+    of a given operation derivatives scheme (shape) and a shift object which
+    can be a None, a single value or a tuple with the given shape.
+    """
+    if shift is None:
+        return lambda s, d, i, j: None
+    if type(shift) is float:
```

Comment:
`if isinstance(shift, float)` 

this way it will work even if eg shift is a numpy.float32 . Besides being more pythonic